[2025-04-20T03:51:53.171+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spacex_ml_training_pipeline.train_model manual__2025-04-20T03:51:52.072847+00:00 [queued]>
[2025-04-20T03:51:53.177+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spacex_ml_training_pipeline.train_model manual__2025-04-20T03:51:52.072847+00:00 [queued]>
[2025-04-20T03:51:53.177+0000] {taskinstance.py:1359} INFO - Starting attempt 1 of 2
[2025-04-20T03:51:53.185+0000] {taskinstance.py:1380} INFO - Executing <Task(BashOperator): train_model> on 2025-04-20 03:51:52.072847+00:00
[2025-04-20T03:51:53.189+0000] {standard_task_runner.py:57} INFO - Started process 248 to run task
[2025-04-20T03:51:53.192+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'spacex_ml_training_pipeline', 'train_model', 'manual__2025-04-20T03:51:52.072847+00:00', '--job-id', '3', '--raw', '--subdir', 'DAGS_FOLDER/spacex_training_dag.py', '--cfg-path', '/tmp/tmpyc8f3nbw']
[2025-04-20T03:51:53.197+0000] {standard_task_runner.py:85} INFO - Job 3: Subtask train_model
[2025-04-20T03:51:53.212+0000] {logging_mixin.py:151} WARNING - /home/***/.local/lib/python3.8/site-packages/***/settings.py:195 DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
[2025-04-20T03:51:53.243+0000] {task_command.py:415} INFO - Running <TaskInstance: spacex_ml_training_pipeline.train_model manual__2025-04-20T03:51:52.072847+00:00 [running]> on host 219f96c9b8b2
[2025-04-20T03:51:53.303+0000] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='gaurav' AIRFLOW_CTX_DAG_ID='spacex_ml_training_pipeline' AIRFLOW_CTX_TASK_ID='train_model' AIRFLOW_CTX_EXECUTION_DATE='2025-04-20T03:51:52.072847+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-04-20T03:51:52.072847+00:00'
[2025-04-20T03:51:53.306+0000] {subprocess.py:63} INFO - Tmp dir root location: /tmp
[2025-04-20T03:51:53.307+0000] {subprocess.py:75} INFO - Running command: ['/bin/bash', '-c', 'python /opt/***/project/train.py']
[2025-04-20T03:51:53.315+0000] {subprocess.py:86} INFO - Output:
[2025-04-20T03:51:55.139+0000] {subprocess.py:93} INFO - 2025/04/20 03:51:55 INFO mlflow.tracking.fluent: Experiment with name 'SpaceX Landing Prediction' does not exist. Creating a new experiment.
[2025-04-20T03:51:55.165+0000] {subprocess.py:93} INFO - 2025/04/20 03:51:55 WARNING mlflow.utils.git_utils: Failed to import Git (the Git executable is probably not on your PATH), so Git SHA is not available. Error: Failed to initialize: Bad git executable.
[2025-04-20T03:51:55.165+0000] {subprocess.py:93} INFO - The git executable must be specified in one of the following ways:
[2025-04-20T03:51:55.166+0000] {subprocess.py:93} INFO -     - be included in your $PATH
[2025-04-20T03:51:55.166+0000] {subprocess.py:93} INFO -     - be set via $GIT_PYTHON_GIT_EXECUTABLE
[2025-04-20T03:51:55.166+0000] {subprocess.py:93} INFO -     - explicitly set via git.refresh(<full-path-to-git-executable>)
[2025-04-20T03:51:55.166+0000] {subprocess.py:93} INFO - 
[2025-04-20T03:51:55.166+0000] {subprocess.py:93} INFO - All git commands will error until this is rectified.
[2025-04-20T03:51:55.166+0000] {subprocess.py:93} INFO - 
[2025-04-20T03:51:55.167+0000] {subprocess.py:93} INFO - This initial message can be silenced or aggravated in the future by setting the
[2025-04-20T03:51:55.167+0000] {subprocess.py:93} INFO - $GIT_PYTHON_REFRESH environment variable. Use one of the following values:
[2025-04-20T03:51:55.167+0000] {subprocess.py:93} INFO -     - quiet|q|silence|s|silent|none|n|0: for no message or exception
[2025-04-20T03:51:55.167+0000] {subprocess.py:93} INFO -     - warn|w|warning|log|l|1: for a warning message (logging level CRITICAL, displayed by default)
[2025-04-20T03:51:55.167+0000] {subprocess.py:93} INFO -     - error|e|exception|raise|r|2: for a raised exception
[2025-04-20T03:51:55.168+0000] {subprocess.py:93} INFO - 
[2025-04-20T03:51:55.168+0000] {subprocess.py:93} INFO - Example:
[2025-04-20T03:51:55.168+0000] {subprocess.py:93} INFO -     export GIT_PYTHON_REFRESH=quiet
[2025-04-20T03:51:55.168+0000] {subprocess.py:93} INFO - 
[2025-04-20T03:51:57.398+0000] {subprocess.py:93} INFO - 2025/04/20 03:51:57 WARNING mlflow.models.model: Model logged without a signature and input example. Please set `input_example` parameter when logging the model to auto infer the model signature.
[2025-04-20T03:52:00.283+0000] {subprocess.py:93} INFO - 2025/04/20 03:52:00 WARNING mlflow.models.model: Model logged without a signature and input example. Please set `input_example` parameter when logging the model to auto infer the model signature.
[2025-04-20T03:52:04.659+0000] {subprocess.py:93} INFO - 2025/04/20 03:52:04 WARNING mlflow.models.model: Model logged without a signature and input example. Please set `input_example` parameter when logging the model to auto infer the model signature.
[2025-04-20T03:52:07.057+0000] {subprocess.py:93} INFO - 2025/04/20 03:52:07 WARNING mlflow.models.model: Model logged without a signature and input example. Please set `input_example` parameter when logging the model to auto infer the model signature.
[2025-04-20T03:52:07.063+0000] {subprocess.py:93} INFO - Training LogisticRegression...
[2025-04-20T03:52:07.064+0000] {subprocess.py:93} INFO - → LogisticRegression logged to MLflow.
[2025-04-20T03:52:07.064+0000] {subprocess.py:93} INFO - Training SVM...
[2025-04-20T03:52:07.064+0000] {subprocess.py:93} INFO - → SVM logged to MLflow.
[2025-04-20T03:52:07.064+0000] {subprocess.py:93} INFO - Training DecisionTree...
[2025-04-20T03:52:07.065+0000] {subprocess.py:93} INFO - → DecisionTree logged to MLflow.
[2025-04-20T03:52:07.065+0000] {subprocess.py:93} INFO - Training KNN...
[2025-04-20T03:52:07.065+0000] {subprocess.py:93} INFO - → KNN logged to MLflow.
[2025-04-20T03:52:07.065+0000] {subprocess.py:93} INFO - Best model saved: LogisticRegression
[2025-04-20T03:52:07.330+0000] {subprocess.py:97} INFO - Command exited with return code 0
[2025-04-20T03:52:07.355+0000] {taskinstance.py:1398} INFO - Marking task as SUCCESS. dag_id=spacex_ml_training_pipeline, task_id=train_model, execution_date=20250420T035152, start_date=20250420T035153, end_date=20250420T035207
[2025-04-20T03:52:07.409+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 0
[2025-04-20T03:52:07.422+0000] {taskinstance.py:2776} INFO - 0 downstream tasks scheduled from follow-on schedule check
